java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java

```diff
@@ -325,6 +342,93 @@ public static void readPrependValues(VariableRegistry registry, String prependFi
         }
     }
 
+    public static void processYaml(VariableRegistry registry, File yamlInputFile, ReaderState state) throws IOException {
+        Yaml yaml = new Yaml();
+        Map<Integer, String> listOutputs = new HashMap<>();
+        Map<Integer, String> listAnalogInputs = new HashMap<>();
+        Map<Integer, String> listEventInputs = new HashMap<>();
+        Map<Integer, String> listSwitchInputs = new HashMap<>();
+        Map<String, Object> yamlData = yaml.load(new FileReader(yamlInputFile));
+        List<Map<String, Object>> data = (List<Map<String, Object>>) yamlData.get("pins");
+        if (data == null) {
+            SystemOut.println("Null yaml for " + yamlInputFile);
+        } else {
+            SystemOut.println(data);
+            Objects.requireNonNull(data, "data");
+            for (Map<String, Object> pin : data) {
+                if (pin.get("id").getClass().getName().equals("java.util.ArrayList")) {
+                    for (int i = 0; i < ((ArrayList)pin.get("id")).size(); i++) {
+                        assignPinName((String)((ArrayList)pin.get("id")).get(i),
+                                (String)pin.get("ts_name"),
+                                (String)((ArrayList)pin.get("class")).get(i),
+                                listOutputs, listAnalogInputs, listEventInputs, listSwitchInputs, state);
+                    }
+                } else {
+                    assignPinName((String)pin.get("id"), (String)pin.get("ts_name"), (String)pin.get("class"), listOutputs, listAnalogInputs, listEventInputs, listSwitchInputs, state);
+                }
+            }
+            registerPins(listOutputs, "output_pin_e_enum", "GPIO_UNASSIGNED", registry);
+            registerPins(listAnalogInputs, "adc_channel_e_enum", "EFI_ADC_NONE", registry);
+            registerPins(listEventInputs, "brain_input_pin_e_enum", "GPIO_UNASSIGNED", registry);
+            registerPins(listSwitchInputs, "switch_input_pin_e_enum", "GPIO_UNASSIGNED", registry);
+        }
+    }
+
+    private static int getMaxValue(Collection<Integer> values) {
+        int result = -1;
+        for (Integer v : values) {
```

Comment:
`return values.stream().max(Integer::compare).get()`

Reply:
I completely removed that function. Did I do it right?

Reply:
https://github.com/rusefi/rusefi/pull/2231/commits/e0e10241a6a3e3af3985149bfc54a9003b9beafe

Reply:
Looks good to me (assuming it compiles and works of course 🙂 )!